lib/rust/ensogl/component/grid-view/src/basic.rs

```diff
@@ -0,0 +1,182 @@
+impl Default for EntryParams {
+    fn default() -> Self {
+        Self {
+            bg_color:    color::Rgba::transparent(),
+            bg_margin:   0.0,
+            font:        text::typeface::font::DEFAULT_FONT.into(),
+            text_offset: 7.0,
+            text_size:   text::Size(14.0),
+            text_color:  default(),
+        }
+    }
+}
+
+
+
+// =============
+// === Entry ===
+// =============
+
+// === EntryData ===
+
+/// An internal structure of [`Entry`], which may be passed to FRP network without risk of memory
+/// leaks.
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub struct EntryData {
+    display_object: display::object::Instance,
+    pub label:      text::Area,
+    pub background: entry_background::View,
+}
+
+impl EntryData {
+    fn new(app: &Application, text_layer: &Option<Layer>) -> Self {
+        let logger = Logger::new("list_view::entry::Label");
+        let display_object = display::object::Instance::new(&logger);
+        let label = app.new_view::<ensogl_text::Area>();
+        let background = entry_background::View::new(&logger);
+        display_object.add_child(&label);
+        display_object.add_child(&background);
+        if let Some(layer) = text_layer {
+            label.add_to_scene_layer(layer);
+        }
+        Self { display_object, label, background }
+    }
+
+    fn update_layout(
+        &self,
+        entry_size: Vector2,
+        bg_margin: f32,
+        text_size: text::Size,
+        text_offset: f32,
+    ) {
+        use entry_background::PADDING_PX;
+        let bg_size = entry_size - Vector2(bg_margin, bg_margin) * 2.0;
+        let bg_size_with_padding = bg_size + Vector2(PADDING_PX, PADDING_PX) * 2.0;
+        self.background.size.set(bg_size_with_padding);
+        self.label.set_position_xy(Vector2(text_offset - entry_size.x / 2.0, text_size.raw / 2.0));
+    }
+}
+
+
+// === Entry ===
+
+/// A [`BasicGridView`] entry - a label with background.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Entry {
+    frp:  EntryFrp<Self>,
+    data: Rc<EntryData>,
+}
+
+impl crate::Entry for Entry {
+    type Model = ImString;
+    type Params = EntryParams;
+
+    fn new(app: &Application, text_layer: &Option<Layer>) -> Self {
+        let data = Rc::new(EntryData::new(app, text_layer));
+        let frp = EntryFrp::<Self>::new();
+        let input = &frp.private().input;
+        let network = frp.network();
+
+        enso_frp::extend! { network
+            bg_color <- input.set_params.map(|params| params.bg_color).on_change();
+            bg_margin <- input.set_params.map(|params| params.bg_margin).on_change();
+            font <- input.set_params.map(|params| params.font.clone_ref()).on_change();
+            text_offset <- input.set_params.map(|params| params.text_offset).on_change();
+            text_color <- input.set_params.map(|params| params.text_color).on_change();
+            text_size <- input.set_params.map(|params| params.text_size).on_change();
```

Comment:
here we fit the line, so it's not so important, but in such cases as these, I normally prefer to write `|t| t.bg_color` instead of `|params| params.bg_color` because the previous word already indicates what we are dealing with (`set_params.map`). Again, not important at all, but shorter code is easier to parsing with eyes.